harness/determined/cli/_util.py

```diff
@@ -100,7 +101,14 @@ def setup_session(args: argparse.Namespace) -> api.Session:
     master_url = args.master or util.get_default_master_address()
     cert = certs.default_load(master_url)
 
-    return api.Session(master_url, args.user, authentication.cli_auth, cert)
+    retry = urllib3.util.retry.Retry(
+        raise_on_status=False,
+        total=5,
+        backoff_factor=0.5,  # {backoff factor} * (2 ** ({number of total retries} - 1))
+        status_forcelist=[502, 503, 504],  # Bad Gateway, Service Unavailable, Gateway Timeout
```

Comment:
Per https://github.com/determined-ai/determined/pull/6853#discussion_r1196864229, I think we want to give this an `allowed_methods`.